framework/src/consumption/lib/redshift/redshift-data.ts

```diff
@@ -0,0 +1,333 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { createHash } from 'crypto';
+import { CustomResource, Duration, RemovalPolicy, Stack, Tags } from 'aws-cdk-lib';
+import { InterfaceVpcEndpoint, InterfaceVpcEndpointAwsService, Peer, Port, SecurityGroup } from 'aws-cdk-lib/aws-ec2';
+import { Effect, IRole, ManagedPolicy, PolicyDocument, PolicyStatement, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { Construct } from 'constructs';
+import { RedshiftDataProps } from './redshift-data-props';
+import { Context, TrackedConstruct, TrackedConstructProps } from '../../../utils';
+import { DsfProvider } from '../../../utils/lib/dsf-provider';
+
+/**
+ * Creates an asynchronous custom resource that handles the execution of SQL using Redshift's Data API. If `vpc` and `vpcSubnets` are passed, this construct would also create the Redshift Data Interface VPC endpoint and configure the custom resource in the same VPC subnet.
+ *
+ * @example
+ * const workgroup = new dsf.consumption.RedshiftServerlessWorkgroup(this, "RedshiftWorkgroup", {
+ *    workgroupName: "redshift-workgroup"
+ * })
+ *
+ * const rsData = workgroup.accessData()
+ * rsData.createDbRole("defaultdb", "engineering")
+ */
+export class RedshiftData extends TrackedConstruct {
+  /**
+   * Custom resource provider. This references the Lambda function that sends the SQL to Redshift via Redshift's Data API
+   */
+  readonly executionProvider: string;
```

Comment:
We only exposes the token of the provider. The Provider itself creates resources that we need to expose in this construct. We have 2 options:
- follow the `S3DataCopy` example and create relevant class members
- expose the `DsfProvider` itself that contains all the resources created